Streams/FileSliceReader.cpp

```diff
@@ -2,15 +2,30 @@
 #include <stdexcept>
 
 FileSliceReader::FileSliceReader(std::string filename, uint64_t startingOffset, uint64_t sliceLength) : 
-	fileStreamReader(filename), startingOffset(startingOffset), sliceLength(sliceLength)
+	fileStreamReader(filename), 
+	startingOffset(startingOffset), 
+	sliceLength(sliceLength)
+{
+	Initialize();
+}
+
+FileSliceReader::FileSliceReader(const FileSliceReader& fileSliceReader) :  
+	fileStreamReader(fileSliceReader.GetFilename()), 
+	startingOffset(fileSliceReader.startingOffset), 
+	sliceLength(fileSliceReader.sliceLength)
+{
+	Initialize();
+}
+
+void FileSliceReader::Initialize() 
```

Comment:
How about InitializeFileStream. Could be used on both the FileSliceReader and FileStreamReader's version of the function.

-Brett

Reply:
The "Initialize" part sounds kind of vague. Also maybe a bit inaccurate now that I think about it, since the fields are already initialized at that point. They're just being error checked. Plus you have the somewhat unrelated operation of seeking.

The code of course makes perfect sense as written. I just feel the name doesn't help to understand what the method really does.

To be fair, the name isn't terrible. This is certainly something we could change post merge if a better name comes to mind.

Reply:
Good point. I think maybe I meant it more as InitializeClass when first writing. I'll just leave it alone for now though.

-Brett